pkg/readiness/ready_tracker.go

```diff
@@ -333,6 +345,32 @@ func (t *Tracker) collectInvalidExpectations(ctx context.Context) {
 	}
 }
 
+func (t *Tracker) trackAssignMetadata(ctx context.Context) error {
+	defer func() {
+		t.assignMetadata.ExpectationsDone()
+		log.V(1).Info("AssignMetadata expectations populated")
+
+		_ = t.constraintTrackers.Wait()
+	}()
+
+	if !*mutation.MutationEnabled {
```

Comment:
nit: not a fan of reaching out to global variables in other packages. In fact, I'm not sure the tracker should be aware of the existence of the `mutation` package at all. Can we change this to be a flag that is passed to the tracker instead?

Reply:
Done